pathfinder/token_network.py

```diff
@@ -155,27 +155,28 @@ def update_balance(
         )
 
     def update_fee(
-        self,
-        channel_id: ChannelId,
-        new_fee: float,
-        signature: Address
+            self,
+            channel_id: ChannelId,
+            new_fee: bytes,
+            signature: bytes
     ):
-        """ Update the channel with a new fee.
-
-        Called by the public interface. """
-
-        # FIXME: Signature on fee update not checked !!
-        # msg = new_fee
-        # signer = from_signature_and_message(signature, msg)
-        # <-- this verifies the signature of the fee-update!
-        signer = signature
+        # Fixme: I need a nonce for replay protection
+        msg = new_fee
+        signer = public_key_to_address(
+            PublicKey.from_signature_and_message(
+                signature,
+                msg,
+                hasher=sha256)
+        )
         participant1, participant2 = self.channel_id_to_addresses[channel_id]
         if is_same_address(participant1, signer):
             sender = participant1
             receiver = participant2
+            new_fee = float(new_fee)
         elif is_same_address(participant2, signer):
             sender = participant2
             receiver = participant1
+            new_fee = float(new_fee)
```

Comment:
Does this work on a `bytes` object? If we stick with `bytes` please specify the internal representation of `new_fee` in a docstring.